x-pack/plugins/asset_manager/public/lib/public_assets_client.test.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { HttpSetupMock } from '@kbn/core-http-browser-mocks';
+import { coreMock } from '@kbn/core/public/mocks';
+import { PublicAssetsClient } from './public_assets_client';
+
+describe('Public assets client', () => {
+  let http: HttpSetupMock = coreMock.createSetup().http;
+
+  beforeEach(() => {
+    http = coreMock.createSetup().http;
+  });
+
+  describe('class instantiation', () => {
+    it('should successfully instantiate', () => {
+      new PublicAssetsClient(http);
+    });
+  });
+
+  describe('getHosts', () => {
+    it('should call the REST API', async () => {
+      const client = new PublicAssetsClient(http);
+      await client.getHosts({ from: 'x', to: 'y' });
+      expect(http.get).toBeCalledTimes(1);
+    });
+
+    it('should include specified "from" and "to" parameters in http.get query', async () => {
+      const client = new PublicAssetsClient(http);
+      await client.getHosts({ from: 'x', to: 'y' });
+      expect(http.get).toBeCalledWith(expect.stringContaining('/api/'), {
```

Comment:
can we test the full api path ?

Reply:
From public client all the way down to … real data? Or to the server client? I think it's a good idea but would like to hear y'all's thoughts on what type of test to use and where to draw the right boundaries here.

Reply:
I meant the string literal `/api/assets/hosts` :)

The server client would already tested by integration tests (added basic test in https://github.com/elastic/kibana/pull/166756). Testing the public client require browser test I think but granted we have coverage for the api I'm fine with unit tests for now

Reply:
Ohh, ok much easier haha. I left it out because it's shared between route and client so it should be changeable without any problems. I can pull that constant in here and use it in the test, though.